tests/integration/test_k8s_with_cluster.py

```diff
@@ -150,8 +147,67 @@ def test_workflow_and_service_management_end_to_end_from_cli(
             delete_cluster_role_binding(workflow_sa_crb)
 
 
+def test_services_from_previous_deployments_are_deleted():
+    job_name = "test-service-cleanup"
+    try:
+        process_one = run(
+            [
+                "bodywork",
+                "deployment",
+                "create",
+                f"--name={job_name}",
+                "--git-repo-url=https://github.com/bodywork-ml/test-single-service-project.git",
+                "--git-repo-branch=test-two-services",
+            ],
+            encoding="utf-8",
+            capture_output=True,
+        )
+        assert process_one.returncode == 0
+        assert "Deployment successful" in process_one.stdout
+
+        sleep(5)
+
+        process_two = run(
+            [
+                "bodywork",
+                "deployment",
+                "create",
+                f"--name={job_name}",
+                "--git-repo-url=https://github.com/bodywork-ml/test-single-service-project.git",
+            ],
+            encoding="utf-8",
+            capture_output=True,
+        )
+        assert process_two.returncode == 0
+        assert "Deployment successful" in process_two.stdout
+        assert (
+            "Removing service: bodywork-test-single-service-project--stage-2 from previous deployment with git-commit-hash" # noqa
+            in process_two.stdout
+        )
+
+        process_three = run(
```

Comment:
Is it worth adding one of these checks - using `bw deployment display` immediately after the first deployment?

Reply:
No I don't think there is any real benefit from doing that. The assertions would fail if the first deployment failed and also if the second deployment does not remove one of the services from the previous deployment.

Reply:
I get you, the test would fail if that service hadn't been created as we check that the service is removed in the output.